File: pet-adoption-be/PetAdoptionApp_PRN221_Group9/BusinessObjects/Adoption.cs
using BusinessObjects.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessObjects
{
    public class Adoption : BaseEntity
    {
        public DateTime? ApplicationDate { get; set; }
        public DateTime? ApprovalDate { get; set; }
        public AdoptionStatus AdoptionStatus { get; set; }
        public string? AdoptionReason { get; set; }
        public string? PetExperience { get; set; }
        public string? Address { get; set; }
        public string? ContactNumber { get; set; }
        public string? Notes { get; set; }
        public string? UserEmail { get; set; }


        //Relationship
        public Guid? UserId {  get; set;}
        public virtual User? User {  get; set;}
        /*public virtual IEnumerable<Pet>? Pets { get; set;}*/
        public Guid? PetId {  get; set; }
        public virtual Pet? Pet { get; set;}
    }
}

File: pet-adoption-fe/app/types/adoption.ts
import { Image } from "./common";
export interface Adoption {
  id: string;
  applicationDate: string;
  approvalDate: string;
  adoptionStatus: number;
  adoptionReason: string;
  petExperience: string;
  address: string;
  contactNumber: string;
  notes: string;
  userEmail: string;
  userId: string;
  petId: string;
  petName: string;
  petImages: Image[] | null | undefined;
}
